server-utils/src/cors.rs

```diff
@@ -201,18 +246,99 @@ pub fn get_cors_header(origin: Option<&str>, host: Option<&str>, allowed: &Optio
 						}
 					})
 					.map(|_| AccessControlAllowOrigin::Value(Origin::parse(origin)))
-					.map(CorsHeader::Ok).unwrap_or(CorsHeader::Invalid)
+					.map(AllowOrigin::Ok).unwrap_or(AllowOrigin::Invalid)
+				},
+			}
+		},
+	}
+}
+
+/// Validates if the `AccessControlAllowedHeaders` in the request are allowed.
+pub fn get_cors_allow_headers(request_headers: &Headers, cors_allow_headers: &AccessControlAllowHeaders) -> AllowHeaders {
+	// Check if the header fields which were sent in the request are required
+	if let AccessControlAllowHeaders::Only(only) = cors_allow_headers {
+		let are_all_allowed = request_headers.iter()
+			.all(|header| {
+				only.contains(&Ascii::new(header.name().to_owned())) ||
+				get_always_allowed_headers().contains(&Ascii::new(header.name()))
```

Comment:
Why do we alloc `always_allowed_headers` on every iteration? Maybe just create it once and cache?